example/lib/go_router_example.dart

```diff
@@ -0,0 +1,345 @@
+import 'package:adaptive_navigation_widget/adaptive_navigation_widget.dart';
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+
+final GlobalKey<NavigatorState> _rootNavigatorKey =
+    GlobalKey<NavigatorState>(debugLabel: 'root');
+final GlobalKey<NavigatorState> _sectionANavigatorKey =
+    GlobalKey<NavigatorState>(debugLabel: 'sectionANav');
+
+void main() {
+  runApp(NestedTabNavigationExampleApp());
+}
+
+/// An example demonstrating how to use nested navigators
+class NestedTabNavigationExampleApp extends StatelessWidget {
+  /// Creates a NestedTabNavigationExampleApp
+  NestedTabNavigationExampleApp({super.key});
+
+  final GoRouter _router = GoRouter(
+    navigatorKey: _rootNavigatorKey,
+    initialLocation: '/a',
+    routes: <RouteBase>[
+      StatefulShellRoute.indexedStack(
+        builder: (
+          final BuildContext context,
+          final GoRouterState state,
+          final StatefulNavigationShell navigationShell,
+        ) =>
+            AdaptiveNavigationShell(navigationShell: navigationShell),
+        branches: <StatefulShellBranch>[
+          // The route branch for the first tab of the bottom navigation bar.
+          StatefulShellBranch(
+            navigatorKey: _sectionANavigatorKey,
```

Comment:
Is this key actually used or can it be removed?

Reply:
removed